File: test/testUtil.py
from util import *


def test_generateEven():
    return generateEven(20)
print (test_generateEven())


def test_getSize():
    var = [2,3,4,2,1]
    var1 = len(var)
    return var1
print (test_getSize())

def test_addTo():
    arr = [2,4,6,8]
    val = 12
    pos= '0'
    return addTo(arr,val,pos)
print (test_addTo())

def test_enqueue():
    arr = [3,5,7,9]
    val = 11
    return enqueue(arr,val)
print (test_enqueue())

def test_dequeue():
    arr= [1,3,5,7,9]
    return dequeue(arr)
print (test_dequeue())

def test_addToObj():
    loc = 0
    val = 'John Doe'
    obj ={loc, val}
   
    return addToObj(obj,loc,val)
print (test_addToObj())


def test_JSsort():
    arr =[43,56,3,2,1,5,66,7,5,4,33,32,23,]
    # arr=['john','alan','ben','sam']
    # arr=['tom','ben','john',8,10,1]
    asc= False
    return JSsort(arr,asc)
print (test_JSsort())


def test_flatten():
    id = 2602
    name = 'Web Tech'
    year =2
    obj = {'id': id, 'name': name, 'year': year}
    return flatten(obj)
print (test_flatten())

File: test/testUtil.js
const assert = require('assert');
const utils = require('../js/util');


function test_generateEven(){
    return utils.generateEven(20);
}
console.log(test_generateEven());


function test_getSize(){
    var arr=[2,4,3,5,6,7,3,2,1,4,0,5];
    // var arr=[]
    return utils.getSize(arr);
}
console.log(test_getSize());


function test_addTo(){
    var val =20;
    var arr =[1,3,4,6,4,3,54,77,6,5];
    var pos = 2;
    return utils.addTo(arr,val,pos)
}
console.log(test_addTo());


function test_enqueue(){
    var arr=[2,4,6,8,10];
    var x = 12;
    return utils.enqueue(arr,x);
}
console.log(test_enqueue());

function test_dequeue(){
    var arr=[5,4,6,7,9];
    return utils.dequeue(arr)
}
console.log(test_dequeue())


function test_addToObj(){
    var obj = {loc, val}
    var val = "John Doe";
    var loc =0;

    return utils.addToObj(obj,loc,val)
}
console.log(test_addToObj())



function test_JSsort(){
    var arr =[100,5,31,2,44,40,7,55,4,18];
    asc= false;
    return utils.JSsort(arr,asc)
}
console.log(test_JSsort())


function test_flatten(){
 var obj = {
    id: 2602,
    name: "Web Tech",
    year: 2
 };
 return utils.flatten(obj);
}
console.log(test_flatten())

// const testFunction = testUtil();
// function testUtil() {
//     return [
//         test_generateEven,
//         test_getSize,
//         test_addTo,
//         test_enqueue,
//         test_dequeue,
//         test_addToObj,
//         test_JSsort,
//         test_flatten
//     ].forEach(func => {
//         func.call();
//     });
// }

